backend/fms_core/resources/transfer.py

```diff
@@ -0,0 +1,187 @@
+    comment = Field(attribute='comment', column_name='Comment')
+
+
+    class Meta:
+        model = Sample
+        import_id_fields = ()
+        fields = (
+            'source_depleted',
+        )
+        excluded = (
+            'container',
+            'sample_kind',
+            'individual',
+            'child_of',
+            'volume_history',
+            'comment',
+        )
+        export_order = (
+            'source_container',
+            'source_container_coordinates',
+            'destination_container',
+            'destination_container_coordinates',
+            'destination_container_name',
+            'destination_container_kind',
+            'destination_parent_container',
+            'destination_parent_container_coordinates',
+            'source_depleted',
+            'volume_used',
+            'transfer_date',
+            'comment',
+        )
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+        skip_rows(dataset, 7)  # Skip preamble
+
+    def import_obj(self, obj, data, dry_run):
+        self.extracted_from = Sample.objects.get(
+            container__barcode=get_normalized_str(data, "Source Container Barcode"),
+            coordinates=get_normalized_str(data, "Source Location Coord"),
+        )
+        # Cast the "Source Depleted" cell to a Python Boolean value and
+        # update the original sample if needed. This is the act of the
+        # extracted sample depleting the original in the process of its
+        # creation.
+        self.extracted_from.depleted = (self.extracted_from.depleted or
+                                        check_truth_like(get_normalized_str(data, "Source Depleted")))
+
+        # Create a process for the current extraction
+        self.process = Process.objects.create(protocol=Protocol.objects.get(name="Transfer"),
+                                              comment="Sample Transfer (imported from template)")
+        
+        super().import_obj(obj, data, dry_run)
+
+    def import_field(self, field, obj, data, is_m2m=False):
+        if field.attribute in ('source_depleted', 'child_of'):
+            # Computed field, skip importing it.
+            return
+
+        if field.attribute == 'destination_container':
+            parent = None;
+            destination_parent_container_barcode = get_normalized_str(data, "Destination Parent Container Barcode")
+
+            if destination_parent_container_barcode:
+                shared_parent_info = dict(
+                    barcode=destination_parent_container_barcode,
+                )
+                try:
+                    parent = Container.objects.get(**shared_parent_info)
+                except Container.DoesNotExist:
+                    raise Exception("Destination parent container does not exist")
+
+            try:
+                obj.container = Container.objects.get(barcode=get_normalized_str(data, "Destination Container Barcode"))
```

Comment:
I would use get_or_create() here instead of the exception handling approach. You can list the params in a dict skipping the missing optional variables.

Reply:
I got the same code conventions from Extraction. I really dont mind changing it, and even prefer using get_or_create, but if so, I'd prefer it to be used uniformly with get_or_create instead of try/except in the rest of the codebase.

Reply:
Ok lets put a Comment here in the code to remember to convert this and the equivalent in extraction to something using get_or_create ?